src/Constraint/ValueProvider/Compound/IntersectionProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+namespace DigitalRevolution\AccessorPairConstraint\Constraint\ValueProvider\Compound;
+
+use DigitalRevolution\AccessorPairConstraint\Constraint\ValueProvider\ValueProvider;
+use phpDocumentor\Reflection\Type;
+use PHPUnit\Framework\MockObject\Generator;
+
+/**
+ * @SuppressWarnings(PHPMD.EvalExpression)
+ */
+class IntersectionProvider implements ValueProvider
+{
+    /** @var Type[] */
+    private array $typehints;
+
+    /** @var array<string, true> */
+    private static array $createdClasses = [];
+
+    /**
+     * @param Type[] $typehints
+     */
+    public function __construct(array $typehints)
+    {
+        $this->typehints = $typehints;
+    }
+
+    /**
+     * Dynamically create a new PHP abstract class with the desired parent class and/or interfaces.
+     * Using PHPUnit Mock Generator to create an instance of the new class and returning it as test value.
+     */
+    public function getValues(): array
+    {
+        $interfaces = [];
+        $extends    = null;
+        foreach ($this->typehints as $typehint) {
+            $fqnsen = (string)$typehint;
+            if (interface_exists($fqnsen)) {
+                $interfaces[] = $fqnsen;
+            } elseif (class_exists($fqnsen)) {
+                $extends = $fqnsen;
+            }
+        }
+
+        $class = '';
+        if ($extends !== null) {
+            $class .= " extends " . $extends;
+        }
+        if (count($interfaces) > 0) {
+            $class .= " implements " . implode(',', $interfaces) . "{}";
+        }
+
+        /** @var class-string $className */
+        $className = "AccesorPairConstraintIntersectionClass" . md5($class);
+        if (isset(self::$createdClasses[$className]) === false) {
```

Comment:
The `self::$createdClasses` static property should be replaceable by `class_exists($className)`.